packages/twenty-front/src/modules/workflow/workflow-variables/hooks/useAvailableVariablesInWorkflowStep.ts

```diff
@@ -1,111 +1,54 @@
-import { getTriggerStepName } from '@/workflow/workflow-variables/utils/getTriggerStepName';
-import isEmpty from 'lodash.isempty';
-import { useRecoilValue } from 'recoil';
 import { isDefined } from 'twenty-shared';
-import { isEmptyObject } from '~/utils/isEmptyObject';
 
 export const useAvailableVariablesInWorkflowStep = ({
   objectNameSingularToSelect,
 }: {
   objectNameSingularToSelect?: string;
 }): StepOutputSchema[] => {
-  const workflowId = useRecoilValue(workflowIdState);
-  const workflow = useWorkflowWithCurrentVersion(workflowId);
   const workflowSelectedNode = useWorkflowSelectedNodeOrThrow();
   const flow = useFlowOrThrow();
+  const { getStepsOutputSchema } = useStepsOutputSchema();
 
-  if (!isDefined(workflow)) {
-    return [];
-  }
-
-  const trigger = flow.trigger;
-  const steps = flow.steps;
+  const steps = flow.steps ?? [];
 
-  const stepDefinition = getStepDefinitionOrThrow({
-    stepId: workflowSelectedNode,
-    trigger,
-    steps,
-  });
-
-  if (
-    !isDefined(stepDefinition) ||
-    stepDefinition.type === 'trigger' ||
-    !isDefined(steps)
-  ) {
-    return [];
-  }
-
-  const previousSteps = [];
+  const previousStepIds: string[] = [];
 
   for (const step of steps) {
     if (step.id === workflowSelectedNode) {
       break;
     }
-    previousSteps.push(step);
+    previousStepIds.push(step.id);
   }
 
-  const result = [];
-
-  const filteredTriggerOutputSchema = filterOutputSchema(
-    trigger?.settings?.outputSchema as OutputSchema | undefined,
-    objectNameSingularToSelect,
-  );
-
-  if (
-    isDefined(trigger) &&
-    isDefined(filteredTriggerOutputSchema) &&
-    !isEmptyObject(filteredTriggerOutputSchema)
-  ) {
-    const triggerIconKey =
-      trigger.type === 'DATABASE_EVENT'
-        ? getTriggerIcon({
-            type: trigger.type,
-            eventName: splitWorkflowTriggerEventName(
-              trigger.settings?.eventName,
-            ).event,
-          })
-        : getTriggerIcon({
-            type: trigger.type,
-          });
-
-    result.push({
-      id: 'trigger',
-      name: isDefined(trigger.name)
-        ? trigger.name
-        : getTriggerStepName(trigger),
-      icon: triggerIconKey,
-      outputSchema: filteredTriggerOutputSchema,
-    });
-  }
-
-  previousSteps.forEach((previousStep) => {
-    const filteredOutputSchema = filterOutputSchema(
-      previousStep.settings.outputSchema as OutputSchema,
-      objectNameSingularToSelect,
-    );
-
-    if (isDefined(filteredOutputSchema) && !isEmpty(filteredOutputSchema)) {
-      result.push({
-        id: previousStep.id,
-        name: previousStep.name,
-        icon: getActionIcon(previousStep.type),
-        outputSchema: filteredOutputSchema,
-      });
-    }
+  const availableStepsOutputSchema: StepOutputSchema[] =
+    getStepsOutputSchema(previousStepIds).filter(isDefined);
+
+  const triggersOutputSchema: StepOutputSchema[] = getStepsOutputSchema([
+    'trigger',
```

Comment:
You can use `TRIGGER_STEP_ID`, too!

```suggestion
    TRIGGER_STEP_ID,
```